.github/workflows/gpuci.yml

```diff
@@ -0,0 +1,41 @@
+name: "GPU CI/CD"
+
+on:
+  pull_request:
+    branches:
+      - 'main'
+      - 'r**'
```

Comment:
This is from NeMo, meaning the workflow will also trigger for PRs targeting any branch that starts with the letter r followed by any characters (for example, we have `r0.3.0`, `r0.4.0`, and `r0.5.0`).

I personally don't think we need to check for this case but wanted to see what others think?

EDIT: It looks like we also have branches called `rywolf/*`, so this could be changed to `r0**`.

Reply:
If a regex works maybe something like `r[0-9]*`? R followed by any number

Reply:
I don't think that would match to the periods in something like `r0.5.0`.

Reply:
But maybe something like `r[0-9].*`?

Reply:
I would recommend two patterns (besides main), and actually good reminder to fix it for NeMo:

* `r[0-9].[0-9].[0-9]` for release branches
* `[rv][0-9].[0-9].[0-9]rc[0-9]` for pre-release branches (like we had with NeMo 2.0.0rc1)

These are rules you could also use for branch protection. I learned the hard way that GH doesn’t use regex but fnmatch, which is quite limited. So we need two rules to cover both cases.